src/modules/azure-azcopy-sync.test.ts

```diff
@@ -0,0 +1,477 @@
+import { TestUtils } from "../tests/test-utils";
+import {
+    AzureAzcopySync,
+    BlobStorageUrlParts,
+    SyncContainerOptions,
+    SyncLocalFileOptions,
+} from "./azure-azcopy-sync";
+import faker from "faker";
+
+// -----------------------------------------------------------------------------------------
+// #region Variables
+// -----------------------------------------------------------------------------------------
+
+let command = "azcopy";
+let spawnSyncArgs = { shell: true, stdio: "inherit" };
+
+// endregion Variables
+
+// -----------------------------------------------------------------------------------------
+// #region Functions
+// -----------------------------------------------------------------------------------------
+
+const _getBlobStorageUrl = (urlParts: BlobStorageUrlParts) => {
+    const path = urlParts.path ?? "";
+    const sasToken = urlParts.sasToken ?? "";
+    return `"https://${urlParts.account}.blob.core.windows.net/${urlParts.container}/${path}${sasToken}"`;
+};
+
+// endregion Variables
+
+// -----------------------------------------------------------------------------------------
+// #region Functions
+// -----------------------------------------------------------------------------------------
+
+describe("AzureAzcopySync", () => {
+    let spawnSyncSpy: jest.SpyInstance;
+    let shellExitSpy: jest.SpyInstance;
+
+    beforeEach(() => {
+        spawnSyncSpy = TestUtils.spyOnSpawnSync();
+        shellExitSpy = TestUtils.spyOnShellExit();
+    });
+
+    // -----------------------------------------------------------------------------------------
+    // #region localFile
+    // -----------------------------------------------------------------------------------------
+
+    describe("localFile", () => {
+        let localFileOptions: SyncLocalFileOptions;
+        let calculatedCommandArgs: string[];
+
+        beforeEach(() => {
+            localFileOptions = {
+                localFilePath: "c:/something/awesome/text.txt",
+                destination: {
+                    account: "test_account",
+                    container: "test_container",
+                    path: "test_path",
+                    sasToken: "?test_sas_token=test",
+                },
+            };
+            calculatedCommandArgs = [
```

Comment:
Could this be renamed to `expectedCommandArgs`?

Reply:
Yes, thanks for this.  I couldn't think of a better name.  `expected` makes more sense than `calculated`